core/src/main/scala/scala/macros/core/Universe.scala

```diff
@@ -81,11 +82,11 @@ trait Universe extends Gensym {
   type Symbol
   type Denotation
 
-  def symName(sym: Symbol)(implicit m: Mirror): Name
-  def denotInfo(denot: Denotation)(implicit m: Mirror): Type
-  def denotName(denot: Denotation)(implicit m: Mirror): Name
-  def denotSym(denot: Denotation)(implicit m: Mirror): Symbol
-  def caseFields(tpe: Type)(implicit m: Mirror): List[Denotation]
+  def symName(sym: Symbol): Name
+  def denotInfo(denot: Denotation): Type
+  def denotName(denot: Denotation): Name
+  def denotSym(denot: Denotation): Symbol
+  def caseFields(tpe: Type): List[Denotation]
```

Comment:
For another issue: `caseFields` is a little overly specific, we can separate it to `constructorFields` and `symbol.isCaseClass`.

Reply:
I opened https://github.com/scalacenter/macros/issues/27